support/helm/helm.go

```diff
@@ -0,0 +1,149 @@
+	// ReleaseName is used to indicate the name of the release to be used
+	// while installing or upgrading the chart identified by Name into the
+	// Namespace
+	ReleaseName string
+	// Version is the helm chart version that should be deployed to the test
+	// infrastructure
+	Version string
+	// Chart is used to indicate the full path of the .tgz artifact of the Helm
+	// chart in case if your tests require you to be able to deploy a packaged
+	// helm chart that is available locally as the tarball
+	Chart string
+	// Mode is used to indicate the mode in which the helm operation is being
+	// performed. These are the first level commands that are processed by the
+	// helm Binary
+	Mode string
+	// Args is used to pass any additional arguments that you might want to pass
+	// for running the helm command in question
+	Args []string
+}
+
+type Manager struct {
+	e *gexe.Echo
+}
+
+type Option func(*Opts)
+
+const (
+	missingHelm = "'helm' command is missing. Please ensure the tool exists before using the helm manager"
+)
+
+func WithName(name string) Option {
+	return func(opts *Opts) {
+		opts.Name = name
+	}
+}
+
+func WithNamespace(namespace string) Option {
+	return func(opts *Opts) {
+		opts.Namespace = namespace
+	}
+}
+
+func WithReleaseName(releaseName string) Option {
+	return func(opts *Opts) {
+		opts.ReleaseName = releaseName
+	}
+}
+
+func WithVersion(version string) Option {
+	return func(opts *Opts) {
+		opts.Version = version
+	}
+}
+
+func WithChart(chart string) Option {
+	return func(opts *Opts) {
+		opts.Chart = chart
+	}
+}
+
+func WithMode(mode string) Option {
+	return func(opts *Opts) {
+		opts.Mode = mode
+	}
+}
+
+func WithArgs(args ...string) Option {
+	return func(opts *Opts) {
+		opts.Args = append(opts.Args, args...)
+	}
+}
+
+func (m *Manager) processOpts(opts ...Option) *Opts {
+	option := &Opts{}
+	for _, op := range opts {
+		op(option)
+	}
+	return option
+}
+
+func (m *Manager) getCommand(opt *Opts) string {
+	commandParts := []string{"helm", opt.Mode}
+	if opt.Mode == "" {
+		panic("Missing helm operation mode. Please use the WithMode option while invoking the Run")
+	}
+	if opt.Name != "" {
+		commandParts = append(commandParts, opt.Name)
+	}
+	if opt.Chart != "" {
+		commandParts = append(commandParts, opt.Chart)
+	} else {
+		commandParts = append(commandParts, opt.ReleaseName)
+	}
+	if opt.Namespace != "" {
+		commandParts = append(commandParts, "--namespace", opt.Namespace)
+	}
+	if opt.Version != "" {
+		commandParts = append(commandParts, "--version", opt.Version)
+	}
+	commandParts = append(commandParts, opt.Args...)
+	return strings.Join(commandParts, " ")
+}
+
+func (m *Manager) Run(opts ...Option) (proc *exec.Proc) {
+	command := m.getCommand(m.processOpts(opts...))
+	log.V(4).InfoS("Running Helm Operation", "command", command)
+	proc = m.e.RunProc(command)
+	return
+}
+
+func NewManager() *Manager {
+	mgr := &Manager{e: gexe.New()}
+	if mgr.e.Prog().Avail("helm") == "" {
+		panic(missingHelm)
```

Comment:
No need to panic.
Since `Run` is the entry point, move the program check there.
Then return an error if prog is missing.

Reply:
All right. That makes sense. Let me change it accordingly.